packages/@aws-cdk/aws-apigateway/lib/integration.ts

```diff
@@ -193,6 +200,10 @@ export class Integration {
     if (options.connectionType === ConnectionType.INTERNET && options.vpcLink !== undefined) {
       throw new Error('cannot set \'vpcLink\' where \'connectionType\' is INTERNET');
     }
+
+    if (options.timeout && (options.timeout.toMilliseconds() < 50 || options.timeout.toMilliseconds() > 29000)) {
```

Comment:
You also need to sure that timeout is not a token before applying this validation

```suggestion
    if (options.timeout && !options.timeout.isUnresolved() && (options.timeout.toMilliseconds() < 50 || options.timeout.toMilliseconds() > 29000)) {
```